pkg/api/write.go

```diff
@@ -246,37 +247,27 @@ func ingest(inserter ingestor.DBInserter, dataParser *parser.DefaultParser) func
 			return false
 		}
 
-		var receivedSamplesCount, receivedMetadataCount int64
-
-		for _, ts := range req.Timeseries {
-			receivedSamplesCount += int64(len(ts.Samples))
-		}
-		receivedMetadataCount += int64(len(req.Metadata))
-
-		metrics.ReceivedSamples.Add(float64(receivedSamplesCount))
 		begin := time.Now()
-
 		numSamples, numMetadata, err := inserter.Ingest(ctx, req)
 		if err != nil {
 			log.Warn("msg", "Error sending samples to remote storage", "err", err, "num_samples", numSamples)
 			http.Error(w, err.Error(), http.StatusInternalServerError)
-			metrics.FailedSamples.Add(float64(uint64(receivedSamplesCount) - numSamples))
-			metrics.FailedMetadata.Add(float64(uint64(receivedMetadataCount) - numMetadata))
+			pgMetrics.IngestorRequests.With(prometheus.Labels{"type": "metric", "kind": "sample_or_metadata", "code": "400"}).Inc()
 			return false
 		}
-
 		duration := time.Since(begin).Seconds()
-		metrics.IngestedSamples.Add(float64(numSamples))
-		metrics.SentMetadata.Add(float64(numMetadata))
-		metrics.SentBatchDuration.Observe(duration)
+		pgMetrics.IngestorRequests.With(prometheus.Labels{"type": "metric", "kind": "sample_or_metadata", "code": "2xx"}).Inc()
```

Comment:
Why `sample_or_metadata`?

Reply:
It's done because `sample_or_metadata`  the function takes the same time for either sample or metadata requests. Its important to say an `or`, since a request can be completely metadata with 0 samples, and vice-versa. This will avoid misinformation.